### PR TITLE
Status callback

### PR DESCRIPTION
**Pull request by cmdcolin**
This is a demonstration of possible status callback

This is somewhat intermediary between progress bar and opaque loading status
#343

It's not completely finished here but I think it is a nice usability add on for large data, curious if the basic concept (basically, attaching a worker.on and worker.off in BaseRpcDriver) is a way to go for this

### CODE REVIEW

**Comment by cmdcolin** [comment]
Note that PileupTrack does not get the downloading index notification but snpcoverage track does, which has to do with making sure that all paths to the bam/cram setup gets initialized with the status callback

**Comment by cmdcolin** [comment]
I'm pretty happy with how this is turning out...if anyone wants to give it a shot/check the code feel free

File: packages/core/rpc/BaseRpcDriver.ts [hunk 1]
```diff
@@ -187,10 +187,32 @@ export default abstract class BaseRpcDriver {
...
+      functionName,
+      { ...filteredAndSerializedArgs, channel },
+      {
+        timeout: 5 * 60 * 1000, // 5 minutes
+        ...options,
+      },
+    )
+    // @ts-ignore
+    if (worker.off) {
+      // @ts-ignore
```
Comment (rbuels) [inline packages/core/rpc/BaseRpcDriver.ts:213]: any way we can clean up these ignores?

Comment (cmdcolin) [reply packages/core/rpc/BaseRpcDriver.ts]: I moved the logic for the status callback into the webworker driver itself to try to handle this issue

File: packages/alignments/src/BamAdapter/BamAdapter.ts [hunk 2]
```diff
@@ -135,18 +135,21 @@ export default class BamAdapter extends BaseFeatureDataAdapter {
...
   getFeatures(
     region: Region & { originalRefName?: string },
-    opts: BaseOptions = {},
+    opts?: BaseOptions,
   ) {
     const { refName, start, end, originalRefName } = region
+    const { signal, statusCallback = () => {} } = opts || {}
     return ObservableCreate<Feature>(async observer => {
       await this.setup(opts)
+      // @ts-ignore
```
Comment (rbuels) [inline packages/alignments/src/BamAdapter/BamAdapter.ts:144]: is this ts-ignore really needed

**Comment by rbuels** [comment]
clear to merge when ready, would be nice to clean up a few more of the ts-ignores but it's also good to go as is if that turns out not to be very feasible.

**Comment by rbuels** [comment]
also check if jbrowse-desktop still works with this change

**Comment by cmdcolin** [comment]
desktop still works, should add status callbacks to electron too

**Comment by cmdcolin** [comment]
electron status callback added now!

File: packages/alignments/src/BamAdapter/BamAdapter.ts (hunk 2 above)
Comment (cmdcolin) [reply packages/alignments/src/BamAdapter/BamAdapter.ts:143]: nope:) removed

**Comment by cmdcolin** [comment]
:clap: :clap:

**Comment by cmdcolin** [comment]
excited to see this land

**Pull request merged**